js/src/sdk/utils/errors/src/formatter.ts

```diff
@@ -15,72 +21,56 @@ interface ErrorDetails {
 }
 
 export const getAPIErrorDetails = (
-  errorKey: string,
-  axiosError: AxiosError<ErrorResponseData>,
-  predefinedError: Record<string, unknown>
+  axiosError: AxiosError<ErrorResponseData>
 ): ErrorDetails => {
+  const statusCode = axiosError.response?.status;
+  const errorCode = statusCode
+    ? API_TO_SDK_ERROR_CODE[statusCode]
+    : COMPOSIO_SDK_ERROR_CODES.BACKEND.UNKNOWN;
+  const predefinedError = BASE_ERROR_CODE_INFO[errorCode];
+
   const defaultErrorDetails = {
     message: axiosError.message,
     description:
-      axiosError.response?.data?.message ||
+      axiosError.response?.data?.error?.message ||
       axiosError.response?.data?.error ||
       axiosError.message,
     possibleFix:
       "Please check the network connection, request parameters, and ensure the API endpoint is correct.",
   };
 
   const metadata = generateMetadataFromAxiosError(axiosError);
-  switch (errorKey) {
-    case SDK_ERROR_CODES.BACKEND.NOT_FOUND:
-    case SDK_ERROR_CODES.BACKEND.UNAUTHORIZED:
-    case SDK_ERROR_CODES.BACKEND.SERVER_ERROR:
-    case SDK_ERROR_CODES.BACKEND.SERVER_UNAVAILABLE:
-    case SDK_ERROR_CODES.BACKEND.RATE_LIMIT:
-      return {
-        message: `${predefinedError.message || axiosError.message} for ${axiosError.config?.baseURL! + axiosError.config?.url!}`,
-        description: (axiosError.response?.data?.message! ||
-          predefinedError.description) as string,
-        possibleFix: (predefinedError.possibleFix! ||
-          defaultErrorDetails.possibleFix) as string,
-        metadata,
-      };
 
-    case SDK_ERROR_CODES.BACKEND.BAD_REQUEST:
-      const validationErrors = axiosError.response?.data?.errors;
-      const formattedErrors = Array.isArray(validationErrors)
-        ? validationErrors
-            .map((err) => JSON.stringify(err as Record<string, unknown>))
-            .join(", ")
-        : JSON.stringify(
-            validationErrors as unknown as Record<string, unknown>
-          );
+  const errorNameFromBE = axiosError.response?.data?.error?.name;
+  const errorTypeFromBE = axiosError.response?.data?.error?.type;
+  const errorMessage = axiosError.response?.data?.error?.message;
 
-      return {
-        message: `Validation Errors while making request to ${axiosError.config?.baseURL! + axiosError.config?.url!}`,
-        description: `Validation Errors: ${formattedErrors}`,
-        possibleFix:
-          "Please check the request parameters and ensure they are correct.",
-        metadata,
-      };
+  const genericMessage = `${errorNameFromBE || predefinedError.message} ${errorTypeFromBE ? `- ${errorTypeFromBE}` : ""} on ${axiosError.config?.baseURL! + axiosError.config?.url!}`;
 
-    case SDK_ERROR_CODES.BACKEND.UNKNOWN:
-    case SDK_ERROR_CODES.COMMON.UNKNOWN:
+  switch (errorCode) {
+    case COMPOSIO_SDK_ERROR_CODES.BACKEND.NOT_FOUND:
```

Comment:
Why are we using status code again only here, it makes sense to use them as fallback but if the type is specified - let's just use that